great_expectations/data_context/store/inline_store_backend.py

```diff
@@ -100,6 +100,10 @@ def _set(self, key: tuple[str, ...], value: Any, **kwargs: dict) -> None:
 
         if resource_type is DataContextVariableSchema.ALL_VARIABLES:
             config_commented_map_from_yaml = yaml.load(value)
+            # NOTE: fluent datasources may be present under both the `fluent_datasources` & `datasources` key
+            # if fluent datasource is part of `datasources` it will attempt to validate using a marshmallow Datasource schema and fail
+            for name in config_commented_map_from_yaml.get("fluent_datasources", {}):  # type: ignore[union-attr]
+                config_commented_map_from_yaml.get("datasources", {}).pop(name, None)  # type: ignore[union-attr,arg-type,call-arg]
```

Comment:
Do we have plans of removing this logic? Forgot where we landed

Reply:
Yes my goal is to remove this logic before resolving the ticket.